### ISSUE TITLE
[Rebased V8 Branch] undefined reference to `PushAllRegistersAndIterateStack'

### ISSUE THREAD

**Issue by lazyparser** [v8-riscv/v8#151]
**Describe the bug**

```
commit 4244a936a48400d14c25acbf126fbbb16c294e15 (HEAD -> riscv64, riscv/riscv64)
Author: Peng Wu <peng.wu@futurewei.com>
Date:   Wed Aug 12 11:58:26 2020 -0700

    Add wasm-js test to CI
```
**To Reproduce**
1. Build with options https://github.com/isrc-cas/PLCT-Toolbox/blob/master/init_v8_all_in_one.sh
2. Run test/program  ( I ran https://github.com/isrc-cas/PLCT-Toolbox/blob/master/v8team/v8_ci_build_test_upload.sh but only the compilation stage affected)
3. fix atomic by add https://github.com/isrc-cas/chromium-v8-build/commit/87d62aae370869df1e76d7dbb8332fb4548d9a6d#diff-552705aa3942b31ff579ed1448590423R237
to bypass #148

**Expected behavior**
build pass

**Errors/Logs**
```
wuwei@plct-build-5:~/play/v8-riscv/v8$ ninja -C out/riscv64.native.debug -j $(nproc)
ninja: Entering directory `out/riscv64.native.debug'
[8/3944] LINK ./cppgc_standalone
FAILED: cppgc_standalone
riscv64-unknown-linux-gnu-g++ -pie -Wl,--fatal-warnings -Wl,--build-id -fPIC -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -Wl,-z,defs -Wl,--as-needed -rdynamic -nodefaultlibs -pie -Wl,--disable-new-dtags -Wl,-O2 -Wl,--gc-sections -o "./cppgc_standalone" -Wl,--start-group @"./cppgc_standalone.rsp"  -Wl,--end-group  -lc -lgcc_s -lm -lrt -ldl -lpthread -latomic
/home/wuwei/opt/riscv/lib/gcc/riscv64-unknown-linux-gnu/10.1.0/../../../../riscv64-unknown-linux-gnu/bin/ld: obj/v8_cppgc_shared/stack.o: in function `.L0 ':
stack.cc:(.text._ZNK4heap4base5Stack15IteratePointersEPNS0_12StackVisitorE+0x12): undefined reference to `PushAllRegistersAndIterateStack'
collect2: error: ld returned 1 exit status
[9/3944] LINK ./cppgc_for_v8_embedders
FAILED: cppgc_for_v8_embedders
riscv64-unknown-linux-gnu-g++ -pie -Wl,--fatal-warnings -Wl,--build-id -fPIC -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -Wl,-z,defs -Wl,--as-needed -rdynamic -nodefaultlibs -pie -Wl,--disable-new-dtags -Wl,-O2 -Wl,--gc-sections -o "./cppgc_for_v8_embedders" -Wl,--start-group @"./cppgc_for_v8_embedders.rsp"  -Wl,--end-group  -lc -lgcc_s -lm -lrt -ldl -lpthread -latomic
/home/wuwei/opt/riscv/lib/gcc/riscv64-unknown-linux-gnu/10.1.0/../../../../riscv64-unknown-linux-gnu/bin/ld: obj/v8_cppgc_shared/stack.o: in function `.L0 ':
stack.cc:(.text._ZNK4heap4base5Stack15IteratePointersEPNS0_12StackVisitorE+0x12): undefined reference to `PushAllRegistersAndIterateStack'
collect2: error: ld returned 1 exit status
[39/3944] CXX obj/test/unittests/cppgc_unittests_sources/gc-invoker-unittest.o
ninja: build stopped: subcommand failed.
```

**Additional context**

you need to resolve the atomic error first, then see this error

**Comment by penguinwu** [comment]
After the issue is resolved, make sure to update the instruction on the wiki on how to build cross-compiled/native build (if any).

**Comment by penguinwu** [comment]
@thomasyonug Is it fixed by #164? if so, please close.

**Comment by thomasyonug** [comment]
Yeah, this issue is fixed.
